cupy/cuda/compiler.py

```diff
@@ -145,6 +145,17 @@ def _get_arch():
         return min(arch, _nvrtc_max_compute_capability)
 
 
+def _get_arch_for_options(arch=None, jitify=False):
+    # This is needed to differentiate between
+    # compute or sm depending on the CUDA version
+    # needed to ensure backwards compatibility with nvrtc
+    if arch is None:
+        arch = _get_arch()
+    if _cuda_hip_version >= 11010 and not jitify:
+        return f'-arch=sm_{arch}'
+    return f'-arch=compute_{arch}'
```

Comment:
Maybe need to return the flag and use it to decide which method to call (`getCUBIN` or `getPTX`)

```suggestion
        return f'-arch=sm_{arch}', 'cubin'
    return f'-arch=compute_{arch}', 'ptx'
```

Reply:
maybe we can directly return the function?
`nvrtc.getPTX` or `nvrtc.getCUBIN`

Reply:
That's also possible but sounds a bit hacky considering the returned value will be used as a hash key.